### ISSUE TITLE
[Feature Request] food information tooltips

### ISSUE THREAD

**Issue by KraXen72** [maruohon/tweakeroo#212]
![image](https://user-images.githubusercontent.com/21956756/113824327-f27f4580-977f-11eb-8a7d-6dd0d66b4540.png)

this is from appleskin, but apple skin provides some more, cheaty features, which i don't really want, so just having this an option in tweakeroo would be great.

**Comment by KraXen72** [comment]
also, i know there is a vanilla tweaks resource pack for this, but it uses a hacky language font replacement approach, and also doesen't display the saturation.

**Comment by maruohon** [comment]
This is far more likely to appear in MiniHUD than in Tweakeroo, as MiniHUD is all kinds of information output and rendering, whereas Tweakeroo is just random crap which usually doesn't involve much rendering.

**Comment by KraXen72** [comment]
fair enough, my only concern is how much is adding another mod gonna slow down my gameplay? i use fabulously optimized modpack with like half the mods removed and added tweakeroo and durability viewer.

**Comment by maruohon** [comment]
If you don't enable any of the other features then MiniHUD basically does almost nothing. And lots of people are using it all the time anyway, it's a pretty lightweight mod. Only some of the overlay renderers can have a real impact on the framerate, and I'm also going to be taking another look at optimizing those. But thus far I've seen almost nobody complain about MiniHUD having a performance impact. I think there is possibly some kind of issue with the despawn sphere or possibly also the other shapes that I need to look into.

**Comment by KraXen72** [comment]
so if i just turn off minihud's rendering when i don't need it and would only keep the food tooltips, it would have no effect on performance? 

also, does tweakeroo lower performance a lot? i haven't really experienced any frame drops but it definitely feels smoother without it in some cases

**Comment by maruohon** [comment]
Okay now I'm going to bring up "placebo effect", because Tweakeroo doesn't really do anything performance heavy.

Probably one of the "worst things" it does is the `handRestockContinuous` feature, which checks the player's hand item every tick and if necessary tries to restock it. But that is also negligible amount of work compared to anything else that the vanilla game does normally anyway. I think the only rendering Tweakeroo currently has are the Flexible Block Placement overlay and the Elytra Camera and Snap Aim overlay. Those are also completely negligible and unnoticeable amount of work compared to any of the rendering the vanilla game does. And if you are not using those features, then that rendering code obviously does not even run.

It's of course possible that I'm forgetting something or there is a bug somewhere, but unless you can show me a profiler result that actually points to something in Tweakeroo, I'm going to say that it's impossible for you to tell when Tweakeroo is installed based on how the game performs. Unless you turn this around and test with some of the options in Tweakeroo that optimize things, or (mostly currently in 1.12.2 and 1.13.2) the options that disable client side light updates, which can gain you a MASSIVE performance boost with large flying machines and stuff, like stone generators that are used for filling up a perimeter.

And as to MiniHUD: yes, if you turn off rendering, then the mod essentially does nothing.

**Comment by KraXen72** [comment]
okay, thank you for detailed explanations. i wish you the best of luck developing these mods, they are really useful.

one thing i noticed, you forget to close a lot of your github issues, maybe take a look at closing some? (like feature requests that were implemented etc)

**Comment by maruohon** [comment]
Most of them are left open on purpose. I'm going to go through the issues when I work on the next port from 1.12.2 to the later versions, and work on and close all relevant issues at that point, when they are actually being resolved for the MC version most people are using, ie. the latest official MC version.

**Comment by Nicolas-GE** [comment]
you can also use the Hunger preview from the [VanillaTweaks](https://vanillatweaks.net/picker/resource-packs/) Resource pack (it's in the Utility tab)

@KraXen72

**Comment by K0-RR** [comment]
@nicolasGEL
> also, i know there is a vanilla tweaks resource pack for this, but it uses a hacky language font replacement approach, and also doesen't display the saturation.

...
